internal/manifests/collector/service.go

```diff
@@ -32,6 +32,7 @@ import (
 // headless and monitoring labels are to differentiate the headless/monitoring services from the clusterIP service.
 const (
 	headlessLabel   = "operator.opentelemetry.io/collector-headless-service"
+	headfulLabel    = "operator.opentelemetry.io/collector-service"
 	monitoringLabel = "operator.opentelemetry.io/collector-monitoring-service"
```

Comment:
This _may_ be a breaking change, but i'm not 100% sure... I think it may be better for us to do something like:
```
operator.opentelemetry.io/collector-service-type: "headless|monitoring|base"
```

or something like that. I think changing the labels on a service are fine as long as you don't change the selector right?

Reply:
It should be OK if users are just relying on the Service Monitors but not sure if they are doing something else.
How about keeping the `operator.opentelemetry.io/collector-monitoring-service` and `operator.opentelemetry.io/collector-headless-service` labels, mark them as deprecated at least in the documentation and use the `operator.opentelemetry.io/collector-service-type` for the `ServiceMonitors`?

Reply:
i would be okay with that... @swiatekm-sumo do you have any opinions?

Reply:
Correct me if I'm wrong, but the problem we're trying to solve here is that we don't want the monitoring ServiceMonitor to select the headless Service, right? In that case, wouldn't it be simplest to add an explicit exclusion to the label selector?

Reply:
This PR tries to solve two different issues:
1. Exclude the headless service from the ServiceMonitor
2. The current `ServiceMonitor` is not finding the `collector` service (so, the Prometheus exporter is not working with that automatically created `ServiceMonitor`). It worked before.

Reply:
Ok, I'm a bit lost with this PR, but this is how I think it should work:

    We have separate ServiceMonitors for Collector metrics and metrics from Prometheus exporter.
    They don't share any metrics, although they can share the same label selector.
    In the above case, the ports should be different.
    Both of them should exclude the headless Service based on its label.

Is there a reason this can't work? @jaronoff97 let me know what you think as well.

Reply:
yes, that's what I think the solution should be too – we would probably have each of the servicemonitor's ports use the proper names too

Reply:
Ok, since we agree, I'll implement the approach suggested by @jaronoff97